github/lib

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+: "${BASH_LIB_DIR:?BASH_LIB_DIR must be set. Please source bash-lib/init before other scripts from bash-lib.}"
+
+function bl_hub_available(){
+    # type instead of which, so it can be stubbed in tests
+    type hub >/dev/null || bl_fail "hub (github cli) binary not found, please install it via your package manager or use bl_hub_download_latest."
+}
+
+function bl_hub_creds_available(){
+    config_file="${HUB_CONFIG:-${HOME}/.config/hub}"
+    [[ -n "${GITHUB_USER:-}" ]] && [[ -n "${GITHUB_TOKEN:-}" ]] && return
+    [[ -e "${config_file}" ]] && return
+    bl_fail "No credentials found for (git)hub please set GITHUB_USER and GITHUB_TOKEN or create ~/.config/hub"
+}
+
+# shellcheck disable=SC2119
```

Comment:
I don't see the context for this, need to be relocated?

Reply:
Good catch, thats left over from a previous iteration.